apps/target_apps_store.py

```diff
@@ -70,26 +70,55 @@ def images_size(self, target: FactoryClient.Target):
 
         return app_image_size
 
-    def copy(self, target: FactoryClient.Target, dst_dir: str):
+    def copy(self, target: FactoryClient.Target, dst_images_dir: str, dst_apps_dir: str):
+        if not self.exist(target):
+            # Try to find an archive file that contains just images
+            if self.exist(target, just_images=True):
+                logger.info('Copying only Apps\' container images...')
+                return self._copy_images(target, dst_images_dir)
+            else:
+                raise Exception('Failed to found images of the given Target {}'.format(target.name))
+
         _, app_image_tar, _ = self.apps_location(target)
 
-        if os.path.exists(dst_dir):
+        if os.path.exists(dst_images_dir):
             # wic image was populated by container images data during LmP build
             # let's remove it and populate with the given images data
             logger.info('Removing existing preloaded app images from the system image')
-            shutil.rmtree(dst_dir)
+            shutil.rmtree(dst_images_dir)
 
-        os.makedirs(dst_dir)
+        os.makedirs(dst_images_dir)
         logger.info('Copying container images of Target Apps; Target: {}, Source: {}, Destination: {}'
-                    .format(target.name, app_image_tar, dst_dir))
-        subprocess.check_call(['tar', '-xf', app_image_tar, '-C', dst_dir])
+                    .format(target.name, app_image_tar, dst_images_dir))
+        subprocess.check_call(['tar', '--strip-components=2', '--exclude=./apps',
+                               '-xf', app_image_tar, '-C', dst_images_dir])
 
-    def apps_location(self, target: FactoryClient.Target):
+        if dst_apps_dir is None:
+            logger.info('Destination compose apps root dir is not specified, skipping Apps copying')
+            return
+
+        if os.path.exists(dst_apps_dir):
+            # wic image was populated by container images data during LmP build
+            # let's remove it and populate with the given images data
+            logger.info('Removing existing preloaded compose apps from the system image')
+            shutil.rmtree(dst_apps_dir)
+
+        os.makedirs(dst_apps_dir)
+        logger.info('Copying Target Apps; Target: {}, Source: {}, Destination: {}'
+                    .format(target.name, app_image_tar, dst_apps_dir))
+        subprocess.check_call(['tar', '--strip-components=2', '--exclude=./images',
```

Comment:
stinks there's not a "--include" option. that would make it a little more future proof

Reply:
do we need to update lmp-image-tools also?

Reply:
> do we need to update lmp-image-tools also?

no, it works as it is, I verified key use-cases in my factory.

Reply:
> stinks there's not a "--include" option. that would make it a little more future proof

I think there is another way to achieve "--include effect", will try it tomorrow.

Reply:
updated to make it "include" not exclude.